imap/mailbox.c

```diff
@@ -4076,7 +4077,9 @@ static int mailbox_update_sieve(struct mailbox *mailbox,
         sdata->isactive = isactive;
         sdata->id = id;
         sdata->name = name;
-        sdata->content = content;
+
+        message_guid_generate(&uuid, content, strlen(content));
```

Comment:
Yeah, this is fine.  We could theoretically read it from the cache datastructure instead of mapping the message again, but this is legit as well.

Reply:
Heh. That didn't even occur to me. I was looking for it in the index record instead.

Reply:
@brong https://github.com/cyrusimap/cyrus-imapd/pull/3487/commits/a6b3a468ed1356b49574217b12ac412002c7fb52